packages/video_player/video_player_avfoundation/ios/Classes/messages.g.m

```diff
@@ -0,0 +1,555 @@
+
+@implementation FLTPlaybackSpeedMessage
++ (instancetype)makeWithTextureId:(nullable NSNumber *)textureId speed:(nullable NSNumber *)speed {
+  FLTPlaybackSpeedMessage *pigeonResult = [[FLTPlaybackSpeedMessage alloc] init];
+  pigeonResult.textureId = textureId;
+  pigeonResult.speed = speed;
+  return pigeonResult;
+}
++ (FLTPlaybackSpeedMessage *)fromMap:(NSDictionary *)dict {
+  FLTPlaybackSpeedMessage *pigeonResult = [[FLTPlaybackSpeedMessage alloc] init];
+  pigeonResult.textureId = dict[@"textureId"];
+  if ((NSNull *)pigeonResult.textureId == [NSNull null]) {
+    pigeonResult.textureId = nil;
+  }
+  pigeonResult.speed = dict[@"speed"];
+  if ((NSNull *)pigeonResult.speed == [NSNull null]) {
+    pigeonResult.speed = nil;
+  }
+  return pigeonResult;
+}
+- (NSDictionary *)toMap {
+  return [NSDictionary
+      dictionaryWithObjectsAndKeys:(self.textureId ? self.textureId : [NSNull null]), @"textureId",
+                                   (self.speed ? self.speed : [NSNull null]), @"speed", nil];
+}
+@end
+
+@implementation FLTPositionMessage
++ (instancetype)makeWithTextureId:(nullable NSNumber *)textureId
+                         position:(nullable NSNumber *)position {
+  FLTPositionMessage *pigeonResult = [[FLTPositionMessage alloc] init];
+  pigeonResult.textureId = textureId;
+  pigeonResult.position = position;
+  return pigeonResult;
+}
++ (FLTPositionMessage *)fromMap:(NSDictionary *)dict {
+  FLTPositionMessage *pigeonResult = [[FLTPositionMessage alloc] init];
+  pigeonResult.textureId = dict[@"textureId"];
+  if ((NSNull *)pigeonResult.textureId == [NSNull null]) {
+    pigeonResult.textureId = nil;
+  }
+  pigeonResult.position = dict[@"position"];
+  if ((NSNull *)pigeonResult.position == [NSNull null]) {
+    pigeonResult.position = nil;
+  }
+  return pigeonResult;
+}
+- (NSDictionary *)toMap {
+  return [NSDictionary
+      dictionaryWithObjectsAndKeys:(self.textureId ? self.textureId : [NSNull null]), @"textureId",
+                                   (self.position ? self.position : [NSNull null]), @"position",
+                                   nil];
+}
+@end
+
+@implementation FLTCreateMessage
++ (instancetype)makeWithAsset:(nullable NSString *)asset
+                          uri:(nullable NSString *)uri
+                  packageName:(nullable NSString *)packageName
+                   formatHint:(nullable NSString *)formatHint
+                  httpHeaders:(nullable NSDictionary<NSString *, NSString *> *)httpHeaders {
+  FLTCreateMessage *pigeonResult = [[FLTCreateMessage alloc] init];
+  pigeonResult.asset = asset;
+  pigeonResult.uri = uri;
+  pigeonResult.packageName = packageName;
+  pigeonResult.formatHint = formatHint;
+  pigeonResult.httpHeaders = httpHeaders;
+  return pigeonResult;
+}
++ (FLTCreateMessage *)fromMap:(NSDictionary *)dict {
+  FLTCreateMessage *pigeonResult = [[FLTCreateMessage alloc] init];
+  pigeonResult.asset = dict[@"asset"];
+  if ((NSNull *)pigeonResult.asset == [NSNull null]) {
```

Comment:
Pigeon fix (so, not for this PR): a simpler solution occurs to me for this type issue than local variables. You can generate a static helper C function at the top of this file and use it for the lookups:

```
NSObject *GetNullableObjectFromDictionary(NSDictionary *dict, NSString *key) {
  NSObject *value = dict[key];
  return value == [NSNull null] ? nil : value;
}
```

Then fromMap is just a series of:
```
pigeonResult.foo = GetNullableObjectFromDictionary(@"foo");
...
```

statements. Shorter, and no type abuse.

Reply:
Good call.  I'll try to do some cleanup on the few pigeon things I saw today.  I'll add that to the list.